pkgs/servers/search/typesense/default.nix

```diff
@@ -0,0 +1,48 @@
+{ lib
+, stdenv
+, fetchurl
+, autoPatchelfHook
+, nixosTests
+}:
+let
+  inherit (stdenv.hostPlatform) system;
+  throwSystem = throw "Unsupported system: ${system}";
+
+  sources = lib.importJSON ./sources.json;
+  platform = sources.platforms.${system} or throwSystem;
```

Comment:
Won't you break evaluation this way on other platforms?

Reply:
Good question, I don't actually know. I copied the pattern from somewhere else in nixpkgs (e.g. https://github.com/NixOS/nixpkgs/blob/2db7b837401df50dd66788827aeaadbdb09a3009/pkgs/misc/urbit/default.nix#L22)

Reply:
please try to evaluate it with like a platform that is not listed aarch64 darwin or x86_64 darwin or i686 linux

Reply:
It refuses evaluation because I set `meta.platforms`. Is that correct or should something else be happening?

```
nix-repl> outputs.legacyPackages.aarch64-darwin.typesense
error: Package ‘typesense-0.24.1’ in /nix/store/3ypa6p1i7s9p2wnjnbwc9fmrb5mkjgkg-source/pkgs/servers/search/typesense/default.nix:43 is not available on the requested hostPlatform:
         hostPlatform.config = "aarch64-apple-darwin"
         package.meta.platforms = [
           "x86_64-linux"
           "aarch64-linux"
           "x86_64-darwin"
         ]
         package.meta.badPlatforms = [ ]
[...]
       , refusing to evaluate.
```

Reply:
That looks correct to me!